### ISSUE TITLE
ARM support

### ISSUE THREAD

**Issue by Gorgeous-Patrick** [CAENTainer/GCC-Images#2]
EECS 281 is thinking about releasing this container image to students. One major roadblock is that it does not support ARM 64 version (MacOS). Reading the commit history I figured out that it was removed intentionally. Was there a particular reason?

**Comment by developStorm** [comment]
This has been released to students at least in F22: https://eecs281staff.github.io/eecs281setup/guides/unified. 

Build ARM native images on x86 machines by using qemu (or attempting some form of cross compilation, which was quite complex when I tried). Currently, building this image natively for x86 already takes hours, and building it in qemu simply takes too much time to be practical. However, it would be feasible now if we had an ARM build machine. Then, we could separately build the image for both platforms natively and publish an umbrella metadata that includes both images.

This, however, doesn’t mean that students can’t use this image on ARM. Docker automatically creates containers based on x86 images using qemu. Students need considerably fewer computational resources compared to compiling GCC and other tools during the build process, making it a more likely solution to compromise runtime speed, although it can be significant.
